### ISSUE TITLE
MD030: use multi-line spacing on paragraphs wrapped across multiple lines

### ISSUE THREAD

**Issue by gioannidis** [markdownlint/markdownlint#507]
## Describe the Enhancement:

I am trying to trigger the `{ol,ul}_multi` spacing rule of [MD030](https://github.com/markdownlint/markdownlint/blob/f97a364ed880a690f509d144641b89ba16dd92e5/docs/RULES.md#md030---spaces-after-list-markers) for lists that:

- Are not nested.
- Have single-paragraph items.
- At least one paragraph spans across multiple lines.

Example:

```markdown
<!--- spaces_after_list_marker.md -->
Raises MD030 error.

-   item that
    is wrapped

-   item 2
```

This raises the following errors:

```text
$ mdl -s test/rule_tests/spaces_after_list_marker_style.rb spaces_after_list_marker.md
spaces_after_list_marker.md:4: MD030 Spaces after list markers
spaces_after_list_marker.md:7: MD030 Spaces after list markers
```

For reference, the style used is [`test/rule_tests/spaces_after_list_marker_style.rb`](https://github.com/markdownlint/markdownlint/blob/f97a364ed880a690f509d144641b89ba16dd92e5/test/rule_tests/spaces_after_list_marker_style.rb).

### Impacted Rules:

- [MD030 - Spaces after list markers](https://github.com/markdownlint/markdownlint/blob/f97a364ed880a690f509d144641b89ba16dd92e5/docs/RULES.md#md030---spaces-after-list-markers)

## Describe the Need:

This affects users adhering to the following style guides:

- Ciro Santilli style guide: [spaces after list marker](https://cirosantilli.com/markdown-style-guide/#spaces-after-list-marker)
- Google style guide: [Nested list spacing](https://google.github.io/styleguide/docguide/style.html#nested-list-spacing)

Both style guides dictate that increased indentation should be used when a paragraph spans across multiple lines in a list item. As a result, these cases fall under the `{ol,ul}_multi` spacing rule, as opposed to the `{ol,ul}_single` rule.

For reference, the [`lib/mdl/styles/cirosantilli.rb`](https://github.com/markdownlint/markdownlint/blob/f97a364ed880a690f509d144641b89ba16dd92e5/lib/mdl/styles/cirosantilli.rb) style is supposed to enforce the [respective style guide](https://cirosantilli.com/markdown-style-guide), but it fails to validate the [example above](#describe-the-enhancement).

## Current Alternative

The current alternative is to explicitly break a multi-line single paragraph into multiple paragraphs:

```markdown
<!--- spaces_after_list_marker_workaround.md -->
Workaround. Triggers no MD030 errors.

-   item that

    is wrapped

-   item 2
```

## Can We Help You Implement This?

Yes, I would be happy to either implement this or at least receive some guidance into understanding the code so that I can implement a custom rule for my own repository.

I have already taken a look at [`lib/mdl/rules.rb`](https://github.com/markdownlint/markdownlint/blob/f97a364ed880a690f509d144641b89ba16dd92e5/lib/mdl/rules.rb#L526-L552). The key line is the following, as it decides whether the multi-line or single-line rule is used:

```ruby
537      srule = items.map { |i| i.children.length }.max > 1 ? 'multi' : 'single'
```

If I understand correctly, the `multi` spacing rule is activated if the list has any children (is nested). I do not understand, though, which part of the code dictates: "_trigger the rule when we have multiple paragraphs_".

What should probably be added is: "_invoke the `multi` spacing rule if at least one item wraps across multiple lines_"

For the record, I have minimal experience in `ruby`; my core languages are `c++` and `python`.
